File: eAnimalShelter/eAnimalShelter.Services/Database/Notification.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using eAnimalShelter.Model.Enums;

namespace eAnimalShelter.Services.Database
{
    public class Notification
    {
        [Key]
        public int NotificationId { get; set; }
        public int? UserId { get; set; }
        [ForeignKey("UserId")]
        public User? User { get; set; }

        // ciljna rola (Client / Volunteer / Admin)
        public int? TargetRoleId { get; set; }
        [ForeignKey("TargetRoleId")]
        public Role? TargetRole { get; set; }

        public string Title { get; set; } = string.Empty;
        public string Message { get; set; } = string.Empty;
        public NotificationType Type { get; set; }
        public bool IsRead { get; set; }
        public DateTime DateSent { get; set; }
        public DateTime? ReadAt { get; set; }
    }

    
}

File: eAnimalShelter/UI/eanimalshelter_desktop/lib/models/notification.dart
enum NotificationType {
  adoption,
  volunteer,
  donation,
  announcement,
  system
}

class NotificationModel {
  int? notificationId;

  int? userId;
  String? userName;

  int? targetRoleId;
  String? targetRoleName;

  String? title;
  String? message;

  int? type;

  bool? isRead;

  DateTime? dateSent;
  DateTime? readAt;

  NotificationModel({
    this.notificationId,
    this.userId,
    this.userName,
    this.targetRoleId,
    this.targetRoleName,
    this.title,
    this.message,
    this.type,
    this.isRead,
    this.dateSent,
    this.readAt,
  });

  factory NotificationModel.fromJson(
    Map<String, dynamic> json,
  ) {
    return NotificationModel(
      notificationId: json["notificationId"],
      userId: json["userId"],
      userName: json["userName"],
      targetRoleId: json["targetRoleId"],
      targetRoleName: json["targetRoleName"],
      title: json["title"],
      message: json["message"],
      type: json["type"],
      isRead: json["isRead"],
      dateSent: json["dateSent"] != null
          ? DateTime.parse(json["dateSent"])
          : null,
      readAt: json["readAt"] != null
          ? DateTime.parse(json["readAt"])
          : null,
    );
  }
}